### ISSUE TITLE
Reproject results in wrong pixel size occasionally

### ISSUE THREAD

**Issue by liuh886** [GlacioHack/geoutils#354]
**Describe the bug**

```
import xdem
import pyproj

fid_fab = 'N62E010_FABDEM_V1-0.tif'
demfab = xdem.DEM(fid_fab).reproject(dst_crs=pyproj.CRS(32633),dst_res=(20,20))
grid = xdem.DEM('us_nga_egm08_25.tif')

demfab
>Driver:               None 
>Opened from file:     None 
>Filename:             None 
>Raster modified since disk load?  True 
>Size:                 2959, 5741
>Number of bands:      1
>Data types:           ('float32',)
>Coordinate System:    ['EPSG:32633']
>NoData Value:         -9999.0
>Pixel Size:           20.0, 20.0
>Upper Left Corner:    238286.29553975418, 6880633.456051373
>Lower Right Corner:   297466.2955397542, 6995453.456051373

grid.reproject(demfab)
>Driver:               None 
>Opened from file:     None 
>Filename:             None 
>Raster modified since disk load?  True 
>Size:                 2959, 5741
>Number of bands:      1
>Data types:           ('float32',)
>Coordinate System:    ['EPSG:32633']
>NoData Value:         -99999
>Pixel Size:           20.00000000000001, 20.0
>Upper Left Corner:    238286.29553975418, 6880633.456051373
>Lower Right Corner:   297466.2955397542, 6995453.456051373
```
The grid has a pixel size of 20.00000000000001 and 20.0, which is wrong.

**To Reproduce**
This is an occasional bug but often happens when dealing with multiple DEMs. For example ['N62E010_FABDEM_V1-0.tif'](https://drive.google.com/file/d/10usaitjrvmhAgywB_3wPOGDGWJZ8iVnT/view?usp=share_link). No clue why it happened.

**Expected behavior**
The pixel size should be exactly the same as dst_ref, (20,20).

**System (please complete the following information):**
 - OS: Windows 10
 - geoutils 0.10

**Comment by rhugonnet** [comment]
Thanks @liuh886. Weird one, indeed. Heard the same from a colleague, so it must happen quite a bit.

I'll have a look to see if it comes from geoutils, or lower (rasterio).

**Comment by liuh886** [comment]
Yes, it happens a lot in my case (2%). I am not that familiar with geoutils, but it looks like from rasterio

**Comment by rhugonnet** [comment]
@liuh886 Fixed in #356, thanks for the reproducible example! Looks like it's from Rasterio, but I'm too lazy to open a "floating point precision" issue right now (those are the worst).

GitHub Actions is down this afternoon so I cannot run CI. Should be able to merge tomorrow once they fix it. :wink:
